src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -344,7 +365,13 @@ private static ImmutableArray<TExtension> CreateExtensionsForAllLanguages(Extens
                 var builder = ImmutableArray.CreateBuilder<TExtension>();
                 foreach (var analyzers in map.Values)
                 {
-                    builder.AddRange(analyzers);
+                    foreach (var analyzer in analyzers)
+                    {
+                        if (includeDuplicates || !builder.Contains(t => t.GetType().Equals(analyzer.GetType())))
```

Comment:
>builder.Contains [](start = 50, length = 16)

It looks like this is O(n^2) where n is the number of extensions. Consider adding all analyzers as before and check for duplicates outside the loop: `return builder.ToImmutable().Distinct(comparer);`

Reply:
The irony that I originally did it this way to avoid the duplicate work of calling distinct afterwards. Good catch.